### PR TITLE
Adiciona autorização com a gem Cancan

### PR DESCRIPTION
**Pull request by luchiago**
Resolve [#5 ](https://github.com/ES2-UFPI/CajehForm-back-end/issues/5)
Adiciona a gem e cria a classe Ability para definir as autorizações de usuário.

### CODE REVIEW

File: app/models/ability.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    user ||= User.new
+    if user.leader
```
Comment (jeduardo824) [inline app/models/ability.rb]: Acho que temos além de Lider, Admins e as autorizações deles são diferentes.

Comment (luchiago) [reply app/models/ability.rb]: No model de usuário está definido apenas o `collab` e o `lider`. Como podemos implementar o admin ?

Comment (jeduardo824) [reply app/models/ability.rb]: The user classes are: Leader (leader: true), collab (collab: true), normal_user(leader: false, collab: false). Leaders are supposed to have the ability to manage everything on the network. Collabs are supposed to be able to post something only. Normal Users are supposed just to consume the network content.

Comment (luchiago) [reply app/models/ability.rb]: Alright, I will make the changes.

**Comment by jeduardo824** [comment]
Resolve esse PR logo

**Pull request closed without merging**